core-bundle/src/Image/Studio/Studio.php

```diff
@@ -0,0 +1,537 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Image\Studio;
+
+use Contao\Config;
+use Contao\ContentModel;
+use Contao\Controller;
+use Contao\CoreBundle\Framework\Adapter;
+use Contao\CoreBundle\Framework\ContaoFramework;
+use Contao\CoreBundle\Image\ImageFactory;
+use Contao\CoreBundle\Image\ImageFactoryInterface;
+use Contao\CoreBundle\Image\PictureFactoryInterface;
+use Contao\CoreBundle\Routing\ScopeMatcher;
+use Contao\FilesModel;
+use Contao\Image\ImageDimensions;
+use Contao\Image\Picture;
+use Contao\Image\PictureConfiguration;
+use Contao\Image\PictureInterface;
+use Contao\LayoutModel;
+use Contao\PageModel;
+use Contao\StringUtil;
+use Contao\Template;
+use Contao\Validator;
+use Psr\Container\ContainerInterface;
+use Symfony\Component\Asset\Context\ContextInterface;
+use Symfony\Component\DependencyInjection\ParameterBag\ParameterBagInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Contracts\Service\ServiceSubscriberInterface;
+use Webmozart\PathUtil\Path;
+
+final class Studio implements ServiceSubscriberInterface
```

Comment:
I like that this uses the service locator pattern because it makes things optimized for performance (doesn't load stuff if not needed).
I also like the configuration design with fluent interfaces 👍